aws/resource_aws_codedeploy_app.go

```diff
@@ -101,52 +110,74 @@ func resourceAwsCodeDeployAppCreate(d *schema.ResourceData, meta interface{}) er
 	// the state file. This allows us to reliably detect both when the TF
 	// config file changes and when the user deletes the app without removing
 	// it first from the TF config.
-	d.SetId(fmt.Sprintf("%s:%s", *resp.ApplicationId, application))
+	d.SetId(fmt.Sprintf("%s:%s", aws.StringValue(resp.ApplicationId), application))
 
 	return resourceAwsCodeDeployAppRead(d, meta)
 }
 
 func resourceAwsCodeDeployAppRead(d *schema.ResourceData, meta interface{}) error {
 	conn := meta.(*AWSClient).codedeployconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
 
 	application := resourceAwsCodeDeployAppParseId(d.Id())
 	log.Printf("[DEBUG] Reading CodeDeploy application %s", application)
 	resp, err := conn.GetApplication(&codedeploy.GetApplicationInput{
 		ApplicationName: aws.String(application),
 	})
 	if err != nil {
-		if codedeployerr, ok := err.(awserr.Error); ok && codedeployerr.Code() == "ApplicationDoesNotExistException" {
+		if isAWSErr(err, codedeploy.ErrCodeApplicationDoesNotExistException, "") {
 			d.SetId("")
+			log.Printf("[WARN] CodeDeploy Application (%s) not found, removing from state", d.Id())
 			return nil
-		} else {
-			log.Printf("[ERROR] Error finding CodeDeploy application: %s", err)
-			return err
 		}
+
+		log.Printf("[ERROR] Error finding CodeDeploy application: %s", err)
+		return err
+	}
+
+	app := resp.Application
+	appName := aws.StringValue(app.ApplicationName)
+
+	appArn := arn.ARN{
```

Comment:
i think we need to include this resource in the list found in `index.html` https://registry.terraform.io/providers/hashicorp/aws/latest/docs#skip_requesting_account_id

Reply:
Forgot about that 😊